suite-native/assets/src/assetsSelectors.ts

```diff
@@ -80,22 +77,19 @@ export const selectDeviceAssetsWithBalances = (state: AssetsRootState & DeviceRo
         const networkAccounts = accountsWithFiatBalance.filter(
             account => account.symbol === networkSymbol,
         );
-        const fiatBalance = networkAccounts
-            .reduce((prev, { symbol, fiatValue }) => {
-                if (symbol === networkSymbol && fiatValue) {
-                    return prev + Number(fiatValue);
-                }
-
-                return prev;
-            }, 0)
-            .toFixed();
-
-        const cryptoValue = sumCryptoBalance(networkAccounts.map(account => account.cryptoValue));
+        const assetBalance = networkAccounts.reduce(
+            (sum, { cryptoValue }) => sum.plus(cryptoValue),
+            new BigNumber(0),
+        );
+        const fiatBalance = networkAccounts.reduce(
```

Comment:
I am quite concern about performance here, this selector is already on my list for being slow and BigNumber is one of biggest perf bottlenecks.

Reply:
`BigNumber` was already used to add crypto amounts and is also used elsewhere when dealing with fiat amounts. I believe it makes sense considering the rounding errors with `number`.

What other solution did you have in mind?

Reply:
Changed back to `Number` based on our discussion in person.